### PR TITLE
uploading supporting files

### PR DESCRIPTION
**Pull request by jyenes**
Slice of code for uploading supporting files

### CODE REVIEW

File: server/xpub-controller/supportingFiles.js [hunk 1]
```diff
@@ -34,6 +80,9 @@ class SupportingFiles {
       if (modified) {
         manuscript.files = filesWithoutSupporting
         manuscript = await manuscript.save()
+        // Required due to pubsweet base-model v1.1.0
+        manuscript = await ManuscriptModel.find(this.manuscriptId, this.user)
+        await manuscript.validate()
```
Comment (hem-brahmbhatt) [inline server/xpub-controller/supportingFiles.js]: I don't think this function exists in develop.

File: server/xpub-controller/supportingFiles.js [hunk 2]
```diff
@@ -9,13 +9,59 @@ class SupportingFiles {
     this.user = user
   }
 
+  async upload(file) {
+    const { stream, filename, mimetype: mimeType } = await file
+    let fileEntity = new FileModel({
+      manuscriptId: this.manuscriptId,
```
Comment (hem-brahmbhatt) [inline server/xpub-controller/supportingFiles.js:15]: `this.manuscriptId` hasn't been set in the constructor, so I think you'll have to pass is in as a parameter for now.

File: server/xpub-controller/supportingFiles.js [hunk 3]
```diff
@@ -9,13 +9,59 @@ class SupportingFiles {
...
+      url: `supporting/${this.manuscriptId}`,
+      filename,
+      type: 'SUPPORTING_FILE',
+      mimeType,
+    })
+    await fileEntity.save()
+    let manuscript = await ManuscriptModel.find(this.manuscriptId, this.user)
+    await manuscript.validate()
```
Comment (hem-brahmbhatt) [inline server/xpub-controller/supportingFiles.js]: This function doesn't exist in develop.

File: server/xpub-controller/supportingFiles.js [hunk 4]
```diff
@@ -9,13 +9,59 @@ class SupportingFiles {
...
+      })
+      stream.on('error', reject)
+      stream.on('end', () => {
+        resolve(Buffer.concat(chunks))
+      })
+    })
+    await fileEntity.updateStatus('UPLOADED')
+    // *unnessicary entity .find()'s after here are required while we are using base-model v1.1.0
+    manuscript = await ManuscriptModel.find(this.manuscriptId, this.user)
+    await manuscript.validate()
```
Comment (hem-brahmbhatt) [inline server/xpub-controller/supportingFiles.js]: This function doesn't exist in develop.

File: server/xpub-controller/supportingFiles.js [hunk 5]
```diff
@@ -9,13 +9,59 @@ class SupportingFiles {
...
+    fileEntity = await FileModel.find(fileId)
+
+    try {
+      await this.storage.putContent(fileEntity, fileContents, {})
+      await fileEntity.updateStatus('STORED')
+      manuscript = await ManuscriptModel.find(this.manuscriptId, this.user)
+      await manuscript.validate()
```
Comment (hem-brahmbhatt) [inline server/xpub-controller/supportingFiles.js]: This function doesn't exist in develop.

File: server/xpub-controller/supportingFiles.js [hunk 6]
```diff
@@ -9,13 +9,59 @@ class SupportingFiles {
...
+    } catch (err) {
+      await fileEntity.updateStatus('CANCELLED')
+      manuscript = await ManuscriptModel.find(this.manuscriptId, this.user)
+      await manuscript.validate()
```
Comment (hem-brahmbhatt) [inline server/xpub-controller/supportingFiles.js]: :shipit:

**Review by hem-brahmbhatt — APPROVED** [review]

**Review by hem-brahmbhatt — APPROVED** [review]

**Review by SemraSh — APPROVED** [review]

**Pull request merged**